### PR TITLE
33 user endpoint

### PR DESCRIPTION
**Pull request by SanderMeister**
added /users/{id} and /users/{id}/pokemon to get a user and to get all the pokemon he or she owns

### CODE REVIEW

File: src/main/kotlin/resource/ItemResource.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import io.ktor.response.respond
+import io.ktor.response.respondText
+import io.ktor.routing.Route
+import io.ktor.routing.get
+import io.ktor.routing.route
+import service.ItemService
+import service.PokemonService
+
+fun Route.item(itemService: ItemService) {
+
```
Comment (SimonBaars) [inline src/main/kotlin/resource/ItemResource.kt]: What does this empty thing do here?

File: src/main/kotlin/resource/PokemonResource.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import io.ktor.request.receiveParameters
+import io.ktor.response.respond
+import io.ktor.response.respondText
+import io.ktor.routing.Route
+import io.ktor.routing.get
+import io.ktor.routing.route
+import service.PokemonService
+
+fun Route.pokemon(pokemonService: PokemonService) {
+
```
Comment (SimonBaars) [inline src/main/kotlin/resource/PokemonResource.kt]: What does this empty thing do here?

File: src/main/kotlin/resource/UserResource.kt [hunk 3]
```diff
@@ -34,6 +37,20 @@ fun Route.user(userService: UserService) {
                 call.respond("You are missing one or multiple parameters")
             }
         }
+
+        get("/{id}") {
+            val id = call.parameters["id"]
+            if (id != null) {
```
Comment (SimonBaars) [inline src/main/kotlin/resource/UserResource.kt]: Not needed.

File: src/main/kotlin/resource/UserResource.kt [hunk 4]
```diff
@@ -34,6 +37,20 @@ fun Route.user(userService: UserService) {
...
+                call.respond(Users.getUser(id.toInt()))
+            }
+        }
+
+        get("/{id}/pokemon") {
+            val id = call.parameters["id"]
+            if (id != null) {
```
Comment (SimonBaars) [inline src/main/kotlin/resource/UserResource.kt]: Not needed.

File: src/main/kotlin/service/ItemService.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import me.sargunvohra.lib.pokekotlin.client.PokeApiClient
+import me.sargunvohra.lib.pokekotlin.model.LocationArea
+import model.*
+import org.jetbrains.exposed.sql.ResultRow
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.transaction
+import utility.PokeApi
+
+class ItemService {
+
```
Comment (SimonBaars) [inline src/main/kotlin/service/ItemService.kt]: What does this empty thing do here?

File: src/main/kotlin/service/PokemonService.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import me.sargunvohra.lib.pokekotlin.client.PokeApiClient
+import me.sargunvohra.lib.pokekotlin.model.LocationArea
+import model.*
+import org.jetbrains.exposed.sql.ResultRow
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.transaction
+import utility.PokeApi
+
+class PokemonService {
+
```
Comment (SimonBaars) [inline src/main/kotlin/service/PokemonService.kt]: What does this empty thing do here?

File: src/main/kotlin/service/user/UserService.kt [hunk 7]
```diff
@@ -14,4 +19,13 @@ class UserService {
     fun registerUser(registrationRequest: UserRegistrationRequest): UserAuthenticationResponse {
         return Registration.registerUser(registrationRequest)
     }
+
+    fun getUserPokemon(userId: Int): List<model.Pokemon> {
+
+        val ownedPokemons = transaction {
+            model.Pokemons.select{ Pokemons.owner eq userId}.map { model.Pokemons.toPokemon(it) }
```
Comment (SimonBaars) [inline src/main/kotlin/service/user/UserService.kt]: Small comment: be consistent in your spacing

**Review by SimonBaars — CHANGES_REQUESTED** [review]

**Comment by loehnertz** [comment]
I fixed the merge conflict and removed the nullability checks.
I approve!

**Review by loehnertz — APPROVED** [review]

**Review by CJStuart — APPROVED** [review]

**Pull request merged**